inference-engine/src/cldnn_engine/ops/custom.cpp

```diff
@@ -248,4 +251,53 @@ void CreateCustomOp(Program& p, const std::shared_ptr<ngraph::Node>& op, CLDNNCu
     p.primitiveIDs[genericLayerName] = prevLayerName;
 }
 
+void CreateGenericPrimitiveOp(Program &p, const std::shared_ptr<ngraph::Node>& op,
+                              InferenceEngine::ILayerExecImpl::Ptr impl) {
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    std::string genericLayerName = layer_type_name_ID(op);
+
+    // TODO: Handle reordering input/output dims... Requires a way to describe input/output layout
+    cldnn::generic_primitive::execute_function f = [genericLayerName, impl](
+            const std::vector<cldnn::event::ptr>& dependent_events,
+            const std::vector<cldnn::memory::ptr>& inputs,
+            const std::vector<cldnn::memory::ptr>& outputs) {
+        cldnn::stream &stream = inputs[0]->get_engine()->get_program_stream();
+
+        // TODO: we don't want to wait for the events, we just want the user to submit onto the queue
+        for (auto& ev : dependent_events) {
+            ev->wait();
+        }
+
+        // TODO: how do we have the user return an event?
+        cldnn::event::ptr ev = stream.create_user_event(false);
+
+        // TODO: Wrap input buffers into RemoteBlobs
```

Comment:
@vladimir-paramuzov There's quite some complication here, not sure how to solve. This callback function `f`, when executed, is at the scope of `cldnn` library. There is no concept of (remote) context at this level, since the (remote) context is managed at the plugin level. Any insight you might have here would be appreciated!

Reply:
Cc @ilyachur

Reply:
@ngaloppo I think you can capture context ptr in this lambda. At the moment context it's not available in `Program` class, so you need to pass it from `CLDNNGraph` to `Program` c-tor probably. 
Another option is to extend remote context implementation to allow creation from cldnn::engine, but I think it's much more complicated way.